### PR TITLE
#4073

### LINKED ISSUE #4073: Segregate even and odd numbers
### Description

Given an array A[], write a function that segregates even and odd numbers. The function should put even numbers first then odd numbers.

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by shrutigrover25**
<hr>

## Description 📜

In a given array, The even numbers will appear first and then the odd numbers. Segregated even and odd numbers in an array.
Fixes #4073

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Doc#umentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [x] C/#CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] Java#Script
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by shrutigrover25** [comment]
Please review my first PR.

File: segregate.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+⭐If left < right then swap arr[left] and arr[right]
+
+IMPLEMENTATION💥💥
+
+// C++ program to segregate even and odd elements of array
+#include <iostream>
+using namespace std;
+ 
+/* Function to swap *a and *b */
+void swap(int *a, int *b);
```
Comment (deeqakkk) [inline segregate.md:26]: complete this function code.

File: segregate.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
+Given an array A[], write a function that segregates even and odd numbers. 
+The functions should put all even numbers first, and then odd numbers.
+
+
+EXAMPLE💥💥 
+
+Input  = {12, 34, 45, 9, 8, 90, 3}
```
Comment (deeqakkk) [inline segregate.md:7]: use a nextline tag here "<br>"

File: segregate.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+⭐Initialize two index variables left and right:  
+            left = 0,  right = size -1 
+⭐Keep incrementing left index until we see an even number.
+⭐Keep decrementing right index until we see an odd number.
+⭐If left < right then swap arr[left] and arr[right]
+
+IMPLEMENTATION💥💥
+
+// C++ program to segregate even and odd elements of array
```
Comment (deeqakkk) [inline segregate.md:21]: nextline tag here as well

File: segregate.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+#include <iostream>
+using namespace std;
+ 
+/* Function to swap *a and *b */
```
Comment (deeqakkk) [inline segregate.md:25]: nextline tag here as well

**Review by deeqakkk — CHANGES_REQUESTED** [review]
Before making a PR make sure, you yourself find your code/content good.
Your content need a good formatting, don't use ir-relevant emoji's
and plz use nextline tag in required statements.
image

**Comment by shrutigrover25** [comment]
I have committed the changes.
Now kindly review.

**Comment by shrutigrover25** [comment]
Ok

**Comment by kashishahuja2002** [comment]
No updates for long, make changes or close the PR. @shrutigrover25 
Please close this PR. @Saviour1001

**Pull request closed without merging**